### PR TITLE
Update template files to comply with linter jobs.

### PR DESCRIPTION
**Pull request by lrcouto**
## Motivation and Context

While working on moving all of the project dependencies to `pyproject.toml` files as part of https://github.com/kedro-org/kedro/issues/4116, our linting CI tasks started pointing out some errors:

```
src/project_dummy/pipeline_registry.py:2:1: UP035 typing.Dict is deprecated, use dict instead
src/project_dummy/pipeline_registry.py:8:29: UP006 [*] Use dict instead of Dict for type annotation
src/project_dummy/pipelines/data_science/nodes.py:2:1: UP035 typing.Dict is deprecated, use dict instead
src/project_dummy/pipelines/data_science/nodes.py:2:1: UP035 typing.Tuple is deprecated, use tuple instead
src/project_dummy/pipelines/data_science/nodes.py:10:48: UP006 [*] Use dict instead of Dict for type annotation
src/project_dummy/pipelines/data_science/nodes.py:10:57: UP006 [*] Use tuple instead of Tuple for type annotation
```

PEP 585 has proposed using Python builtin type hints instead of `Dict`, `Tuple`, and `Set` from the `typing` library, to simplify and avoid duplication. This did not appear as an linter issue until the dependencies were moved, perhaps due to the way Ruff deals with package resolution.

The linting tasks also accused some errors on the formatting of Cookiecutter placeholder lines on pyproject.toml itself, mostly demanding that they be put in double quotes.

Separating this PR from https://github.com/kedro-org/kedro-starters/pull/246 to keep the two issues separate and more organized. https://github.com/kedro-org/kedro-starters/pull/246 would depend on this one to be closed.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Assigned myself to the PR
- [ ] Added tests to cover my changes

### CODE REVIEW

File: astro-airflow-iris/{{ cookiecutter.repo_name }}/pyproject.toml [hunk 1]
```diff
@@ -8,7 +8,7 @@ readme = "README.md"
 dynamic = ["dependencies", "version"]
 
 [project.scripts]
-{{ cookiecutter.repo_name }} = "{{ cookiecutter.python_package }}.__main__:main"
+"{{ cookiecutter.repo_name }}" = "{{ cookiecutter.python_package }}.__main__:main"
```
Comment (merelcht) [inline astro-airflow-iris/{{ cookiecutter.repo_name }}/pyproject.toml:11]: Do the quotes not change any of the behaviour? We don't have these in the base template either.

**Review by merelcht — COMMENTED** [review]
LGTM! 👍 

Can you just double check that adding the double quotes to the cookiecutter placeholders doesn't change the behaviour?

File: astro-airflow-iris/{{ cookiecutter.repo_name }}/pyproject.toml (hunk 1 above)
Comment (noklam) [reply astro-airflow-iris/{{ cookiecutter.repo_name }}/pyproject.toml:11]: I have the same question. Is this a correct fix, it looks like `ruff` is treating this as `pyproject.toml` but it is using jinja syntax that ruff will not understand.

Comment (lrcouto) [reply astro-airflow-iris/{{ cookiecutter.repo_name }}/pyproject.toml:11]: It does not change the behavior from what I'm seeing here. I've tested it by creating a new project with a starter and using this branch as the `--checkout` argument. It resolves the repo name correctly and `kedro run` works as well.

Comment (merelcht) [reply astro-airflow-iris/{{ cookiecutter.repo_name }}/pyproject.toml:11]: I tested it locally as well and it does seem to work fine. I specifically checked that the resulting `pyproject.toml` file looked the same as when using the released version.

**Review by merelcht — APPROVED** [review]

**Review by ElenaKhaustova — APPROVED** [review]
LGTM, thank you!

**Pull request merged**